packer/conf/hooks/environment

```diff
@@ -3,30 +3,26 @@ set -eu
 
 echo '+++ :house_with_garden: Setting up the environment'
 
-[[ -f ~/.docker/config.json ]] && rm ~/.docker/config.json
-[[ -f ~/.dockercfg ]] && rm ~/.dockercfg
+source ~/cfn-env
 
- # handle docker hub authentication
-if [[ -n "${DOCKER_HUB_AUTH:-}" ]] ; then
-  echo "DOCKER_HUB_AUTH is no longer supported, use DOCKER_HUB_USER, DOCKER_HUB_PASSWORD and DOCKER_HUB_EMAIL."
-  exit 1
-elif [[ -n "${DOCKER_HUB_USER:-}" && -n "${DOCKER_HUB_PASSWORD:-}" && -n "${DOCKER_HUB_EMAIL:-}" ]] ; then
-  echo "~~~ Authenticating with :docker: as ${DOCKER_HUB_USER}"
-  docker login --username="${DOCKER_HUB_USER}" --password="${DOCKER_HUB_PASSWORD}" --email="${DOCKER_HUB_EMAIL}"
-elif [[ -n "${DOCKER_HUB_USER:-}" && -n "${DOCKER_HUB_PASSWORD:-}" ]] ; then
-  echo "Missing DOCKER_HUB_EMAIL."
+if [[ -n "$BUILDKITE_SECRETS_BUCKET" ]] && [[ -z "$BUILDKITE_SECRETS_KEY" ]] ; then
+  echo "A SecretsBucket was provided, but no encryption key via BUILDKITE_SECRETS_KEY."
   exit 1
 fi
 
-eval `ssh-agent -s`
+SSH_KEY_NAME="${SSH_KEY_NAME:-id_rsa_github}"
+SSH_KEY_URL="s3://${BUILDKITE_SECRETS_BUCKET}/$SSH_KEY_NAME"
+
+if ! github_ssh_key=$(env -i aws s3 cp --sse-c --sse-c-key "$BUILDKITE_SECRETS_KEY" "$SSH_KEY_URL") ; then
+  echo "Failed to download and decrypt $SSH_KEY_URL"
+fi
+
+eval "$(ssh-agent -s)"
 
 # startup an ssh-agent with the ssh-key
 if [[ -n "${SSH_KEY_URL:-}" ]] ; then
-  echo "~~~ Downloading :github: ssh key"
-  TMPKEY=$(mktemp /tmp/buildkite.XXXXXXXXXX)
-  trap "rm $TMPKEY" exit
-  env -i aws s3 cp ${SSH_KEY_URL:-} $TMPKEY
-  ssh-add $TMPKEY
+  echo "~~~ Downloading github ssh key from $SSH_KEY_URL"
```

Comment:
The download has already been attempted here, the message implies it's about to happen.
You could wrap the download in a function which outputs the key or exits non-zero on failure, rather than stashing it in a temporary variable up above.